src/scep/Client/client.py

```diff
@@ -114,21 +113,21 @@ def get_crl(self, identity, identity_private_key, serial_number, identifier=None
 
         return self._pki_operation(identity=identity, identity_private_key=identity_private_key, envelope=envelope, message_type=MessageType.GetCRL, cacaps=cacaps, ca_certs=ca_certs)
 
-    def enrol(self, csr, identity, identity_private_key, identifier=None):
+    def enroll(self, csr, identity, identity_private_key, cacaps):
         """Perform a PKCSReq operation by submitting a CSR to the SCEP service."""
-        cacaps = self.get_ca_capabilities(identifier=identifier)
-        ca_certs = self.get_ca_certs(identifier=identifier)
-        envelope = PKCSPKIEnvelopeBuilder().encrypt(csr.to_der(), cacaps.strongest_cipher())
-        transaction_id = hex_digest_for_data(data=csr.public_key.to_der(), algorithm='sha1')
+        cacaps = Capabilities(cacaps)
+        ca_certs = CACertificates([identity])
+        envelope = PKCSPKIEnvelopeBuilder().encrypt(csr.to_der(), 'aes256')
+        transaction_id = hex_digest_for_data(data=csr.public_key.to_der(), algorithm='sha256')
         return self._pki_operation(identity=identity, identity_private_key=identity_private_key, envelope=envelope, message_type=MessageType.PKCSReq, cacaps=cacaps, ca_certs=ca_certs, transaction_id=transaction_id)
 
-    def _pki_operation(self, identity, identity_private_key, envelope, message_type, cacaps, ca_certs, transaction_id=None):
+    def _pki_operation(self, identity, identity_private_key, envelope, message_type, cacaps, ca_certs, transaction_id=None, parse=False):
         """Perform a PKIOperation using the PKI Envelope given."""
         envelope = envelope.add_recipient(ca_certs.recipient)
 
         envelope, key, iv = envelope.finalize()
 
-        signer = Signer(identity, identity_private_key, cacaps.strongest_signature_algorithm())
+        signer = Signer(identity, identity_private_key, 'sha256')
```

Comment:
does `cacaps.strongest_signature_algorithm()` not work?

Reply:
It's probably a different algorithm thatn `sha256`

Reply:
It turns out that we are using our Foxpass generated CA to create a client certificate which won't work because we need a private key to generate it.

That is why there is an `Cypher text size does not match error`